test/unit/taggedTemplateLiterals.spec.ts

```diff
@@ -0,0 +1,117 @@
+import * as util from "../util";
+
+const testCases = [
+    {
+        callExpression: "func``",
+        joinAllResult: "",
+        joinRawResult: "",
+    },
+    {
+        callExpression: "func`hello`",
+        joinAllResult: "hello",
+        joinRawResult: "hello",
+    },
+    {
+        callExpression: "func`hello ${1} ${2} ${3}`",
+        joinAllResult: "hello 1 2 3",
+        joinRawResult: "hello   ",
+    },
+    {
+        callExpression: "func`hello ${(() => 'iife')()}`",
+        joinAllResult: "hello iife",
+        joinRawResult: "hello ",
+    },
+    {
+        callExpression: "func`hello ${1 + 2 + 3} arithmetic`",
+        joinAllResult: "hello 6 arithmetic",
+        joinRawResult: "hello  arithmetic",
+    },
+    {
+        callExpression: "func`begin ${'middle'} end`",
+        joinAllResult: "begin middle end",
+        joinRawResult: "begin  end",
+    },
+    {
+        callExpression: "func`hello ${func`hello`}`",
+        joinAllResult: "hello hello",
+        joinRawResult: "hello ",
+    },
+    {
+        callExpression: "func`hello \\u00A9`",
+        joinAllResult: "hello ©",
+        joinRawResult: "hello \\u00A9",
+    },
+    {
+        callExpression: "func`hello $ { }`",
+        joinAllResult: "hello $ { }",
+        joinRawResult: "hello $ { }",
+    },
+    {
+        callExpression: "func`hello { ${'brackets'} }`",
+        joinAllResult: "hello { brackets }",
+        joinRawResult: "hello {  }",
+    },
+    {
+        callExpression: "func`hello \\``",
+        joinAllResult: "hello `",
+        joinRawResult: "hello \\`",
+    },
+    {
+        callExpression: "obj.func`hello ${'propertyAccessExpression'}`",
```

Comment:
Looks like it always calls functions with global context:

```ts
const obj = {
    func(strings: TemplateStringsArray, ...expressions: any[]) {
        console.log(this === obj); // `true` in JS, `false` in Lua
    },
};

obj.func`hello`;
```

```lua
local obj = {func = function(self, strings, ...)
    print(self == obj)
end}
obj.func(_G, { -- should be `obj`
    "hello",
    raw = {"hello"},
})
```

Also there is a case where `obj` isn't pure:

```ts
getObj()["func"]`hello`;
```

In call expressions it's handled like that:

```lua
(function()
    local ____TS_self = getObj(_G)
    return ____TS_self.func(____TS_self, {
        "hello",
        raw = {"hello"},
    })
end)()
```

Reply:
Good find! Took a bit to understand what to do here, I've added `transformContextualCallExpression` in an attempt to preserve this left hand side behaviour to call-like expressions which includes these template literals.